ledger/src/blocktree_processor.rs

```diff
@@ -412,77 +412,174 @@ pub fn verify_ticks(
+    pub num_entries: usize,
+    pub num_txs: usize,
+}
+
+impl ConfirmationProgress {
+    pub fn new(last_entry: Hash) -> Self {
+        Self {
+            last_entry,
+            ..Self::default()
+        }
     }
 }
 
-pub fn verify_and_process_entries(
+pub fn confirm_slot(
+    blocktree: &Blocktree,
     bank: &Arc<Bank>,
-    entries: &[Entry],
-    last_entry: &Hash,
-    slot_full: bool,
-    tick_hash_count: &mut u64,
+    timing: &mut ConfirmationTiming,
+    progress: &mut ConfirmationProgress,
+    skip_verification: bool,
     transaction_status_sender: Option<TransactionStatusSender>,
     entry_callback: Option<&ProcessCallback>,
     recyclers: &VerifyRecyclers,
-) -> result::Result<(u64, u64), BlocktreeProcessorError> {
-    verify_ticks(bank, entries, slot_full, tick_hash_count)?;
+) -> result::Result<(), BlocktreeProcessorError> {
+    let slot = bank.slot();
 
-    datapoint_debug!("verify-batch-size", ("size", entries.len() as i64, i64));
-    let mut verify_total = Measure::start("verify_and_process_entries");
-    let mut entry_state = entries.start_verify(last_entry, recyclers.clone());
+    let (entries, num_shreds, slot_full) = {
+        let mut load_elapsed = Measure::start("load_elapsed");
+        let load_result = blocktree
+            .get_slot_entries_with_shred_info(slot, progress.num_shreds)
+            .map_err(BlocktreeProcessorError::FailedToLoadEntries);
+        load_elapsed.stop();
+        if load_result.is_err() {
+            timing.fetch_fail_elapsed += load_elapsed.as_us();
+        } else {
+            timing.fetch_elapsed += load_elapsed.as_us();
+        }
+        load_result
+    }?;
+
+    let num_entries = entries.len();
+    let num_txs = entries.iter().map(|e| e.transactions.len()).sum::<usize>();
+    trace!(
+        "Fetched entries for slot {}, num_entries: {}, num_shreds: {}, num_txs: {}, slot_full: {}",
+        slot,
+        num_entries,
+        num_shreds,
+        num_txs,
+        slot_full,
+    );
 
-    if entry_state.status() == EntryVerificationStatus::Failure {
-        warn!("Ledger proof of history failed at slot: {}", bank.slot());
-        return Err(BlockError::InvalidEntryHash.into());
+    if !skip_verification {
+        let tick_hash_count = &mut progress.tick_hash_count;
+        verify_ticks(bank, &entries, slot_full, tick_hash_count).map_err(|err| {
+            warn!(
+                "{:#?}, slot: {}, entry len: {}, tick_height: {}, last entry: {}, last_blockhash: {}, shred_index: {}, slot_full: {}",
+                err,
+                slot,
+                num_entries,
+                bank.tick_height(),
+                progress.last_entry,
+                bank.last_blockhash(),
+                num_shreds,
+                slot_full,
+            );
+            err
+        })?;
     }
 
-    let mut processing_elapsed = Measure::start("processing_elapsed");
+    let mut verify_total = Measure::start("verify_and_process_entries");
```

Comment:
Maybe we should make this an option based on `skip_verification` as well, and only use it if `skip_verification` is not set